### PR TITLE
DM-164 Orchestrate application 

### PR DESCRIPTION
**Pull request by KochTobi**
Adds the application logic

### CODE REVIEW

File: src/main/groovy/life/qbic/samplestatus/reporter/commands/ReportSinceInstant.groovy [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  void run() {
+    log.info("Gathering updated samples...")
+    List<Result<SampleUpdate, Exception>> updatedSamples = limsQueryService.getUpdatedSamples(getTimePoint())
+    log.info("Found ${updatedSamples.size()} updated samples.")
+    log.info("Processing updates...")
+    updatedSamples.stream()
+            .filter(Result::isOk)
+            .map(Result::getValue)
+            .peek(it -> log.info("\tupdating $it"))
+            .forEach(statusReporter::reportSampleStatusUpdate)
```
Comment (sven1103) [inline src/main/groovy/life/qbic/samplestatus/reporter/commands/ReportSinceInstant.groovy]: What is the exception handling here? I.e. when the sample status update was not successful?

**Review by sven1103 — COMMENTED** [review]
Question regarding error handling :)

File: src/main/groovy/life/qbic/samplestatus/reporter/commands/ReportSinceInstant.groovy (hunk 1 above)
Comment (KochTobi) [reply src/main/groovy/life/qbic/samplestatus/reporter/commands/ReportSinceInstant.groovy]: The exception handling happens in the ReporterApp

Comment (KochTobi) [reply src/main/groovy/life/qbic/samplestatus/reporter/commands/ReportSinceInstant.groovy]: https://github.com/qbicsoftware/sample-status-reporter/pull/8/files#diff-a6ab529c19bba792c839bd3b1916ba590cf529ac08dead45af54f26a899c2768R45

**Review by sven1103 — APPROVED** [review]
LGTM, i have no objections

File: README.md [hunk 2]
```diff
@@ -1,154 +1,41 @@
...
+| `SAMPLE_TRACKING_AUTH_USER`         | The username for the sample tracking service                           |
+| `SAMPLE_TRACKING_LOCATION_ENDPOINT` | The endpoint to list all locations. This does not contain the base url |
+| `SAMPLE_TRACKING_LOCATION_USER`     | The sample tracking user currently using the application               |
+| `SAMPLE_TRACKING_URL`               | The base URL for the sample tracking service                           |
+| `USER_DB_DIALECT`                   | The database dialect of the user database                              |
+| `USER_DB_DRIVER`                    | The database driver for the user database                              |
+| `USER_DB_HOST`                      | The URL to the host of the user database                               |
+| `USER_DB_NAME`                      | The database name                                                      |
+| `USER_DB_USER_NAME`                 | The database user name                                                 |
+| `USER_DB_USER_PW`                   | The database user password                                             |
```
Comment (sven1103) [inline README.md:23]: 🎉 awesome

File: src/main/groovy/life/qbic/samplestatus/reporter/commands/ReportSinceInstant.groovy [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  ReportSinceInstant(LimsQueryService limsQueryService, SampleStatusReporter statusReporter, UpdateSearchService updateSearchService) {
+    this.limsQueryService = limsQueryService
+    this.statusReporter = statusReporter
+    this.updateSearchService = updateSearchService
+    if (!timePoint) {
+      timePoint = defaultTimePoint
+    }
+  }
+
+  private Instant defaultTimePoint = updateSearchService.getLastUpdateSearchTimePoint().orElse(Instant.now().minus(1, ChronoUnit.DAYS))
```
Comment (sven1103) [inline src/main/groovy/life/qbic/samplestatus/reporter/commands/ReportSinceInstant.groovy]: does this work? can you guarantee that this is executed after the updateSearchService is available to the instance?

**Review by sven1103 — CHANGES_REQUESTED** [review]
Only one concern with the variable instanciation, else all good

File: src/main/groovy/life/qbic/samplestatus/reporter/commands/ReportSinceInstant.groovy (hunk 3 above)
Comment (KochTobi) [reply src/main/groovy/life/qbic/samplestatus/reporter/commands/ReportSinceInstant.groovy]: Yes. I tested it locally and this is called from the constructor so the updateSearchService is available at that time 😄.
And as it is only called after the `updateSearchService` is instantiated this works.

**Review by sven1103 — APPROVED** [review]
Great

File: src/main/groovy/life/qbic/samplestatus/reporter/commands/ReportSinceInstant.groovy (hunk 3 above)
Comment (KochTobi) [reply src/main/groovy/life/qbic/samplestatus/reporter/commands/ReportSinceInstant.groovy]: It changed it as I am unsure how the order of execution is here and whether it is always the same

**Pull request merged**